[docs] Refactor "Selectors" (#4799)
DevExpress/testcafe: docs/articles/documentation/reference/test-api/selector/sibling.md

---
layout: docs
title: Selector.sibling Method
permalink: /documentation/reference/test-api/selector/sibling.html
---
# Selector.sibling Method

Selects sibling ***elements***.

## Syntax

### sibling

```text
Selector().sibling() → Selector
```

> Finds all sibling ***elements*** of all nodes in the matched set.

Reviewer: ```suggestion
Finds the sibling ***elements*** of all nodes in the matched set.
```

### sibling(index)

```text
Selector().sibling(index) → Selector
```

> Finds all sibling ***elements*** of all nodes in the matched set and filters them by index.

Reviewer: See the comment below

Argument | Type   | Description
-------- | ------ | --------------
`index`  | Number | The zero-based index. Elements are indexed as they appear in their parents' `childNodes` collections. If `index` is negative, the index is counted from the end of the matched set.

### sibling(cssSelector)

```text
Selector().sibling(cssSelector) → Selector
```

> Finds all sibling ***elements*** of all nodes in the matched set and filters them by a CSS selector.

Reviewer: ```suggestion
Finds all sibling ***elements*** of all nodes in the matched set and uses a CSS selector to filter them.
```

Argument      | Type   | Description
------------- | ------ | --------------
`cssSelector` | String | The CSS selector string used to filter child elements.

### sibling(filterFn, dependencies)

```text
Selector().sibling(filterFn [, dependencies]) → Selector
```

> Finds all sibling ***elements*** of all nodes in the matched set and filters them by a predicate.

Reviewer: ```suggestion
Finds the sibling ***elements*** of all nodes in the matched set uses a predicate to filter them.
```
